lld/MachO/ObjC.cpp

```diff
@@ -1072,12 +1080,25 @@ void ObjcCategoryMerger::collectAndValidateCategoriesData() {
           tryGetSymbolAtIsecOffset(catBodyIsec, catLayout.klassOffset);
       assert(classSym && "Category does not have a valid base class");
 
-      InfoInputCategory catInputInfo{catListCisec, catBodyIsec, off};
+      InfoInputCategory catInputInfo{catListCisec, catBodyIsec, off,
+                                     inputCategoryIndex++};
```

Comment:
To avoid unnecessary copy,
```suggestion
categoryMap[classSym].emplace_back(catListCisec, catBodyIsec, off,
                                     ++inputCategoryIndex);